app/coffee/DocArchiveManager.coffee

```diff
@@ -60,26 +66,50 @@ module.exports = DocArchive =
 	unarchiveDoc: (project_id, doc_id, callback)->
 		logger.log project_id: project_id, doc_id: doc_id, "getting doc from s3"
 		try
-			options = DocArchive.buildS3Options(true, project_id+"/"+doc_id)
+			options = DocArchive.buildS3Options(project_id+"/"+doc_id)
 		catch e
 			return callback e
-		request.get options, (err, res, lines)->
+		options.json = true
+		request.get options, (err, res, doc)->
 			if err? || res.statusCode != 200
 				logger.err err:err, res:res, project_id:project_id, doc_id:doc_id, "something went wrong unarchiving doc from aws"
 				return callback new Errors.NotFoundError("Error in S3 request")
-			if !(lines instanceof Array)
-				logger.err err:err, res:res, project_id:project_id, doc_id:doc_id, lines:lines, "doc lines from aws are not in array format, likely not JSON parsable"
-				return callback(new Error("Error unpacking doc"))
-			MongoManager.upsertIntoDocCollection project_id, doc_id.toString(), {lines}, (err) ->
-				return callback(err) if err?
-				logger.log project_id: project_id, doc_id: doc_id, "deleting doc from s3"
-				request.del options, (err, res, body)->
-					if err? || res.statusCode != 204
-						logger.err err:err, res:res, project_id:project_id, doc_id:doc_id, "something went wrong deleting doc from aws"
-						return callback new Errors.NotFoundError("Error in S3 request")
-					callback()
+			DocArchive._s3DocToMongoDoc doc, (error, mongo_doc) ->
+				return callback(error) if error?
+				MongoManager.upsertIntoDocCollection project_id, doc_id.toString(), mongo_doc, (err) ->
+					return callback(err) if err?
+					logger.log project_id: project_id, doc_id: doc_id, "deleting doc from s3"
+					request.del options, (err, res, body)->
+						if err? || res.statusCode != 204
+							logger.err err:err, res:res, project_id:project_id, doc_id:doc_id, "something went wrong deleting doc from aws"
+							return callback new Errors.NotFoundError("Error in S3 request")
+						callback()
+
+	_s3DocToMongoDoc: (doc, callback = (error, mongo_doc) ->) ->
+		mongo_doc = {}
+		if doc.schema_v == 1 and doc.lines?
+			mongo_doc.lines = doc.lines
+			if doc.ranges?
+				mongo_doc.ranges = RangeManager.jsonRangesToMongo(doc.ranges)
+		else if doc instanceof Array
+			mongo_doc.lines = doc
+		else
+			return callback(new Error("I don't understand the doc format in s3"))
+		return callback null, mongo_doc
+
+	_mongoDocToS3Doc: (doc, callback = (error, s3_doc) ->) ->
+		json = JSON.stringify({
+			lines: doc.lines
```

Comment:
do we want to null check doc.lines and doc.ranges to ensure they have been passed through Fully and correctly?